### PR TITLE
Feature/646 no item level data

### PR DESCRIPTION
**Pull request by jtreuting**
This sets the minimum school year for item level data to be shown.  Set to 2016 right now because we have item level data for 2016-17 in UAT environment.

When looking at older years the Items By POionts earned button will be disabled with hover title text explaining why and the Student Responses menu option will be the same.

### CODE REVIEW

File: webapp/src/main/webapp/src/app/assessments/assessments.component.ts [hunk 1]
```diff
@@ -160,6 +163,13 @@ export class AssessmentsComponent implements OnInit {
       this.updateFilterOptions();
     });
 
+    this.userService.getCurrentUser().subscribe(user => {
+      console.log(user);
+
+      this.minimumItemDataYear = user.configuration.minItemDataYear;
+      console.log(this.minimumItemDataYear);
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/assessments/assessments.component.ts]: please remove console logs prior to merge

File: webapp/src/main/webapp/src/app/assessments/menu/menu-action.builder.ts [hunk 2]
```diff
@@ -74,6 +75,16 @@ export class MenuActionBuilder {
       this.router.navigate(commands, { relativeTo: this.route });
     }).bind(this);
 
+    responsesAction.isDisabled = ((actionable) => {
+      return !hasItemLevelData(actionable);
+    }).bind(this);
+
+    responsesAction.tooltip = ((actionable) => {
+      return responsesAction.isDisabled(actionable)
+        ? this.translateService.instant('messages.no-item-level-data')
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/assessments/menu/menu-action.builder.ts]: Hm, maybe should be in `labels.menu.disabled-responses-message` for consistency

File: webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html [hunk 3]
```diff
@@ -64,9 +64,12 @@
         <h3 class="gray-darkest mb-sm">
           {{'labels.groups.results.assessment.exams.title' | translate}}
 
-          <a *ngIf="assessmentExam.assessment.isInterim" class="btn btn-default btn-xs pull-right" (click)="viewItemsByPoints(true)">
+          <button *ngIf="assessmentExam.assessment.isInterim"
+                  [disabled]="!hasItemLevelData"
+                  title="{{ hasItemLevelData ? '' : 'messages.no-item-level-data' | translate }}"
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html]: Oh I see why you put it in messages now.

File: webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html [hunk 4]
```diff
@@ -64,9 +64,12 @@
         <h3 class="gray-darkest mb-sm">
           {{'labels.groups.results.assessment.exams.title' | translate}}
 
-          <a *ngIf="assessmentExam.assessment.isInterim" class="btn btn-default btn-xs pull-right" (click)="viewItemsByPoints(true)">
+          <button *ngIf="assessmentExam.assessment.isInterim"
+                  [disabled]="!hasItemLevelData"
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html]: Hm, I wonder if the `isInterim` check should now be part of the `hasItemLevelData` check as to not confuse users why sometimes it's hidden and other times it's disabled.

File: webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html [hunk 5]
```diff
@@ -165,7 +168,7 @@ <h3 class="gray-darkest mb-sm">
             <span>{{'labels.export.items-by-points-earned' | translate}}</span>
           </button>
         </h3>
-        <p-dataTable [value]="filteredAssessmentItems" tableStyleClass="table table-striped table-hover" [tableStyle]="{'table-layout':'auto'}" expandableRows="true" rowExpandMode="single">
+        <p-dataTable *ngIf="hasItemLevelData" [value]="filteredAssessmentItems" tableStyleClass="table table-striped table-hover" [tableStyle]="{'table-layout':'auto'}" expandableRows="true" rowExpandMode="single">
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html]: I don't think we can get here if the button is disabled.

File: webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.ts [hunk 6]
```diff
@@ -55,6 +55,10 @@ export class AssessmentResultsComponent implements OnInit {
   pointColumns: ItemPointField[];
   showItemsByPoints: boolean = false;
 
+  get hasItemLevelData(): boolean {
+    return this._assessmentExam.exams.some(x => x.schoolYear > this.minimumItemDataYear);
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.ts]: Should this be `>=` ?  Either way we should document if the min year is inclusive or not.

File: webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.ts [hunk 7]
```diff
@@ -215,15 +226,18 @@ export class AssessmentResultsComponent implements OnInit {
...
       this.loadAssessmentItems(this.assessmentExam.assessment.id).subscribe(assessmentItems => {
-        this.pointColumns = this.examCalculator.getPointFields(assessmentItems);
 
-        this._assessmentItems = assessmentItems;
-        this.filteredAssessmentItems = this.filterAssessmentItems(assessmentItems);
+        let numOfScores = assessmentItems.reduce((x, y) => x + y.scores.length, 0);
 
-        this.examCalculator.aggregateItemsByPoints(this.filteredAssessmentItems);
-        this.showItemsByPoints = true;
+        if (numOfScores != 0) {
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.ts]: Does something bomb out if you don't do this check?  Or just saving some processing here?

File: webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.ts [hunk 8]
```diff
@@ -322,7 +336,8 @@ export class AssessmentResultsComponent implements OnInit {
     let builder = this.actionBuilder.newActions();
 
     if (!this._assessmentExam.assessment.isSummative) {
-      builder.withResponses(exam => exam.id, exam => exam.student);
+      console.log(this.minimumItemDataYear);
```
Comment (bgsavage) [inline webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.ts]: nit :)

File: webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html (hunk 4 above)
Comment (jtreuting) [inline webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html]: hmmm, not sure.  I think it's different when it' always not available for all summatives

File: webapp/src/main/webapp/src/app/assessments/assessments.component.ts (hunk 1 above)
Comment (jtreuting) [inline webapp/src/main/webapp/src/app/assessments/assessments.component.ts]: doh, missed one,  will do

File: webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html (hunk 5 above)
Comment (jtreuting) [inline webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.html]: yep, you are right.  left over from initial path

File: webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.ts (hunk 7 above)
Comment (jtreuting) [inline webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.ts]: damn, again left over from initial path forward before chanigng direction

Comment (bgsavage) [inline webapp/src/main/webapp/src/app/assessments/results/assessment-results.component.ts]: I figured as much :)

**Pull request merged**